rpc/swap.go

```diff
@@ -44,22 +46,44 @@ func NewSwapService(
 	}
 }
 
+// PastSwapInfo contains a swap ID and its start and end time.
+type PastSwapInfo struct {
+	ID        string    `json:"id"`
+	StartTime time.Time `json:"startTime"`
+	EndTime   time.Time `json:"endTime"`
+}
+
 // GetPastIDsResponse ...
 type GetPastIDsResponse struct {
-	IDs []string `json:"ids"`
+	IDs []*PastSwapInfo `json:"ids"`
```

Comment:
`PastSwaps` for the field name?  These are not IDs anymore.

It would be helpful if the request struct, response struct, RPC method name, and Go method names lined up.

I don't know the answer, but this is probably a good time to ask: Do we want this function at the current time? The implementation is pulling up the entire swap info data structure from disk already to create this list. Given that the reply is a local host transfer, we are not saving a measurable amount of resources by reducing the data before sending it. At some point, especially if we have memory constrained clients, we'll probably want some solution, but now might be too early to optimize. The desired client solution might just be a paging system layered on top of a method that returns the full data structures.

Reply:
do you mean to return all past swap info in this method, instead of just the IDs and start/end time? I think this is useful for quickly seeing your past swaps and checking on their status individually based on time, you might not want to see every swap's full info.

Reply:
Yes, but the client (or `jq` if you are using curl) can be used to filter the fields that you want to see. I was just thinking if we wanted to start simple and only implement one of the variations, returning the full version is probably more useful, and given the current implementation, it has a negligible performance difference. With either version, we may eventually need a paging or time based filter in the request, but that is something that can easily be added later, and adding it can even be done without breaking existing clients.